server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java

```diff
@@ -109,83 +115,213 @@ public void testInvalidKeystoreFile() throws Exception {
+            .stream().findFirst().get();
+        final Environment environment = internalCluster().getInstance(Environment.class);
+        final AtomicReference<AssertionError> reloadSettingsError = new AtomicReference<>();
+        final int initialReloadCount = mockReloadablePlugin.getReloadCount();
+        final char[] password = randomAlphaOfLength(12).toCharArray();
+        writeEmptyKeystore(environment, password);
+        final CountDownLatch latch = new CountDownLatch(1);
+        client().admin()
+            .cluster()
+            .prepareReloadSecureSettings()
+            // No filter should try to hit all nodes
+            .setNodesIds(Strings.EMPTY_ARRAY)
+            .setSecureStorePassword(new SecureString(password))
+            .execute(new ActionListener<NodesReloadSecureSettingsResponse>() {
+                @Override
+                public void onResponse(NodesReloadSecureSettingsResponse nodesReloadResponse) {
+                    reloadSettingsError.set(new AssertionError("Nodes request succeeded when it should have failed", null));
+                    latch.countDown();
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    assertThat(e, instanceOf(ElasticsearchException.class));
+                    assertThat(e.getMessage(),
+                        containsString("Secure settings cannot be updated cluster wide when TLS for the transport layer is not enabled"));
+                    latch.countDown();
+                }
+            });
+        latch.await();
+        if (reloadSettingsError.get() != null) {
+            throw reloadSettingsError.get();
+        }
+        //no reload should be triggered
+        assertThat(mockReloadablePlugin.getReloadCount(), equalTo(initialReloadCount));
+    }
+
+    public void testReloadLocalNodeWithPasswordWithoutTLSSucceeds() throws Exception {
+        final Environment environment = internalCluster().getInstance(Environment.class);
+        final AtomicReference<AssertionError> reloadSettingsError = new AtomicReference<>();
+        final char[] password = randomAlphaOfLength(12).toCharArray();
+        writeEmptyKeystore(environment, password);
+        final CountDownLatch latch = new CountDownLatch(1);
+        client().admin()
+            .cluster()
+            .prepareReloadSecureSettings()
+            .setNodesIds("_local")
+            .setSecureStorePassword(new SecureString(password))
+            .execute(new ActionListener<NodesReloadSecureSettingsResponse>() {
+                @Override
+                public void onResponse(NodesReloadSecureSettingsResponse nodesReloadResponse) {
+                    try {
+                        assertThat(nodesReloadResponse, notNullValue());
+                        final Map<String, NodesReloadSecureSettingsResponse.NodeResponse> nodesMap = nodesReloadResponse.getNodesMap();
+                        assertThat(nodesMap.size(), equalTo(1));
```

Comment:
Tiny nit: you can use `hasSize`.

```suggestion
                        assertThat(nodesMap, hasSize(1));
```

Reply:
Actually there is no hasSize for  `Matcher<java.util.Map<? extends K,? extends V>>`

Reply:
Ah, there's `aMapWithSize()`. Don't know if that's an improvement 🤷‍♂